File: backend/src/main/java/com/intellidesk/api_key/ApiKey.java
package com.intellidesk.api_key;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("api_key")
public class ApiKey {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long workspaceId;
    private Long userId;
    private String name;
    private String keyPrefix;
    private String keyHash;
    private String scope;
    private LocalDateTime expiresAt;
    private LocalDateTime lastUsedAt;
    private String status;
    private LocalDateTime createdAt;
    private LocalDateTime revokedAt;
}

File: frontend/src/types/apiKey.ts
/**
 * API Key domain types.
 *
 * These mirror the Phase 6 backend contract exactly:
 *  - List/detail responses NEVER include fullKey.
 *  - Only the CREATE response includes the one-time fullKey.
 *  - Status comes from backend (ACTIVE / EXPIRED / REVOKED via effectiveStatus).
 */
export type ApiKeyScope = 'READ';

export type ApiKeyStatus = 'ACTIVE' | 'REVOKED';
export type ApiKeyEffectiveStatus = 'ACTIVE' | 'EXPIRED' | 'REVOKED';

export interface ApiKey {
  id: number;
  workspaceId: number;
  name: string;
  keyPrefix: string;
  scope: ApiKeyScope;
  status: ApiKeyStatus;
  effectiveStatus: ApiKeyEffectiveStatus;
  expiresAt?: string;
  lastUsedAt?: string;
  createdAt: string;
  revokedAt?: string;
}

export interface CreateApiKeyRequest {
  name: string;
  scope?: ApiKeyScope;
  expiresAt?: string;
}

/**
 * 创建成功返回的唯一一次 fullKey（plaintext one-time）。
 * 只允许存在 create-success 的短生命周期 runtime state。
 */
export interface CreateApiKeyResult {
  id: number;
  workspaceId: number;
  name: string;
  keyPrefix: string;
  scope: ApiKeyScope;
  fullKey: string;
  expiresAt?: string;
  createdAt: string;
}